### PR TITLE
Add webview support and some clean up

### PR DESCRIPTION
**Pull request by Vindaar**
Closes https://github.com/brentp/nim-plotly/issues/31 by adding a `-d:webview` flag to show the plot using webview instead of a browser.

Note that this is currently completely blocking, the program only continues once the user closes the webview window. I've tried to make it unblocking using webview's `loop` directly (which in principle works), but sometimes resulted in problems when trying to store several images one after another (which works fine with webview!).
I'll try again soonish, but didn't want to delay basic support.

I also went ahead and did some cleanup:
- moved `newPlot`, `add` to `plotly_sugar`
- removed the examples from the main file, since that is covered well by our examples
- made `parseTraces` just rely on procs from the `json` module
- restructured the code in and around the `hasThreadSupport` for (what I hope is) better clarity

### CODE REVIEW

File: src/plotly/plotly_sugar.nim [hunk 1]
```diff
@@ -3,6 +3,19 @@ import sugar
...
+proc newPlot*(xlabel = "", ylabel = "", title = ""): Plot[float64] =
+  ## create a plot with sane default layout.
+  result = Plot[float64]()
+  result.traces = newSeq[Trace[float64]]()
+  result.layout = Layout(title: title, width: 600, height: 600,
+                         xaxis: Axis(title: xlabel),
+                         yaxis: Axis(title: ylabel),
+                         autosize: false)
+
+proc add*[T](p: Plot[T], d: Trace[T]) =
```
Comment (brentp) [inline src/plotly/plotly_sugar.nim]: can we remove this? the API already has a lot of surface area. If we keep adding little conveniences like this, it'll be at the expense of being able to quickly grok the usage.

Comment (Vindaar) [reply src/plotly/plotly_sugar.nim]: Perfectly happy to remove that. I wasn't even aware of its existence to be honest, when I realized it was still using a nil check.

Comment (brentp) [reply src/plotly/plotly_sugar.nim]: ok. that was probably me. :(

**Comment by brentp** [comment]
thanks! this is pretty cool.

**Pull request merged**